src/main/java/graphql/servlet/internal/ApolloSubscriptionProtocolHandler.java

```diff
@@ -58,7 +58,8 @@ public void onMessage(HandshakeRequest request, Session session, WsSessionSubscr
                     subscriptions,
                     message.id,
                     input.getQueryInvoker().query(input.getInvocationInputFactory().create(
-                        input.getGraphQLObjectMapper().getJacksonMapper().convertValue(message.payload, GraphQLRequest.class)
+                        input.getGraphQLObjectMapper().getJacksonMapper().convertValue(message.payload, GraphQLRequest.class),
+                        (HandshakeRequest) session.getUserProperties().get(HandshakeRequest.class.getName())
```

Comment:
I'm not familiar with the handshake request, but you're assuming a certain name here. How is it supposed to be used? So how can you be certain that object will be available with that name?

Reply:
https://github.com/graphql-java/graphql-java-servlet/blob/79c6ae21ffd8e0f4e73d1d675a1c6bf2d53467c5/src/main/java/graphql/servlet/GraphQLWebsocketServlet.java#L110-L111

Because it's provided here

Reply:
Ok, clear. Thanks